merlin_models/data/synthetic.py

```diff
@@ -99,9 +104,9 @@ def read_schema(cls, path: Union[str, Path]) -> Schema:
             os.path.join(str(path), "schema.json") if os.path.isdir(str(path)) else str(path)
         )
         if _schema_path.endswith(".pb") or _schema_path.endswith(".pbtxt"):
-            return Schema().from_proto_text(_schema_path)
+            TensorflowMetadata.from_from_proto_text(_schema_path).to_merlin_schema()
```

Comment:
I think this should be `from_proto_text_file`? Also not sure this works with the way that method is written, since I think the current implementation wants the directory and assumes the file is called `schema.pbtxt` unless you pass a different value of the `filename` param. Now wondering if this is covered by tests too...